homeassistant/components/logbook/__init__.py

```diff
@@ -739,85 +796,87 @@ class LazyEventPartialState:
+        self.time_fired_minute: int = self._row.time_fired.minute
+        self._event_data_cache = event_data_cache
 
     @property
-    def domain(self):
+    def domain(self) -> str | None:
         """Return the domain for the state."""
         if self._domain is None:
+            assert self.entity_id is not None
             self._domain = split_entity_id(self.entity_id)[0]
         return self._domain
 
     @property
-    def attributes_icon(self):
+    def attributes_icon(self) -> str | None:
         """Extract the icon from the decoded attributes or json."""
-        if self._attributes:
-            return self._attributes.get(ATTR_ICON)
         result = ICON_JSON_EXTRACT.search(
-            self._row.shared_attrs or self._row.attributes
+            self._row.shared_attrs or self._row.attributes or ""
         )
-        return result and result.group(1)
+        return result.group(1) if result else None
 
     @property
-    def data_entity_id(self):
+    def data_entity_id(self) -> str | None:
         """Extract the entity id from the decoded data or json."""
         if self._event_data:
             return self._event_data.get(ATTR_ENTITY_ID)
 
         result = ENTITY_ID_JSON_EXTRACT.search(self._row.event_data)
-        return result and result.group(1)
+        return result.group(1) if result else None
 
     @property
-    def data_domain(self):
-        """Extract the domain from the decoded data or json."""
-        if self._event_data:
-            return self._event_data.get(ATTR_DOMAIN)
-
-        result = DOMAIN_JSON_EXTRACT.search(self._row.event_data)
-        return result and result.group(1)
+    def attributes_friendly_name(self) -> str | None:
+        """Extract the friendly name from the decoded attributes or json."""
+        result = FRIENDLY_NAME_JSON_EXTRACT.search(
+            self._row.shared_attrs or self._row.attributes or ""
+        )
+        return result.group(1) if result else None
 
     @property
-    def attributes(self):
-        """State attributes."""
-        if self._attributes is None:
-            source = self._row.shared_attrs or self._row.attributes
-            if source == EMPTY_JSON_OBJECT or source is None:
-                self._attributes = {}
-            else:
-                self._attributes = json.loads(source)
-        return self._attributes
+    def data_domain(self) -> str | None:
+        """Extract the domain from the decoded data or json."""
+        result = DOMAIN_JSON_EXTRACT.search(self._row.event_data)
+        return result.group(1) if result else None
 
     @property
-    def data(self):
+    def data(self) -> dict[str, Any]:
         """Event data."""
         if not self._event_data:
-            if self._row.event_data == EMPTY_JSON_OBJECT:
-                self._event_data = {}
+            source: str = self._row.event_data
+            if event_data := self._event_data_cache.get(source):
```

Comment:
Will this need updating if it becomes it's own table ?

Reply:
Thankfully it won't need refactoring, since it is an in memory cache for the life of the api call. It doesn't care where the data comes from.

The other PR takes care of getting the data into the right place.